lib/r10k/module/git.rb

```diff
@@ -81,6 +88,11 @@ def validate_ref(desired, default)
         msg << "Could not determine desired ref"
       end
 
+      if default_override
+        msg << "or resolve the default branch override '%{default_override}',"
+        vars[:default_override] = default_override
+      end
+
       if default
         msg << "or resolve default ref '%{default}'"
         vars[:default] = default
```

Comment:
I won't let me comment on the actual line, but do we think the "no default provided" message below will make sense if there's a default override but no actual default?

Reply:
The docs make clear that there is a "normal default" and a "default override". This is also the language the puppetfile uses for the options themselves.

Reply:
The whole message becomes: "Could not resolve control repo branch or resolve the default branch override 'override', and no default provided." I think it sounds okay.

Reply:
okey!